index.bs

```diff
@@ -315,6 +315,30 @@ For example,
 is more readable than
 <code highlight="js">window.scrollBy(50, 0)</code>.
 
+<h3 id="promises">Use Promises</h3>
+
+Asynchronous APIs should generally be designed using promises
+rather than callback functions,
+because APIs using promises are generally easier to use
+than APIs using callback functions.
+
+Furthermore, you should carefully consider whether an API might need
+to be asynchronous before making it a synchronous API.
+An API might need to be asynchronous if:
+  * some implementations may (in at least some cases) wish to prompt the user
+    or ask the user for permission before allowing use of the API,
+  * implementations may need to consider information that might be stored
+    on disk in order to compute the result,
+  * implementations may need to interact with the network before returning
+    the result, or
+  * implementations may wish to do work on another thread before returning
+    the result.
```

Comment:
Maybe add "or communicate with an out-of-process subsystem" since that case is pretty common these days.